### PR TITLE
[log-shipper] add extraLabels validation

### LINKED ISSUE #7740: [log-shipper] If extraLabels name starts with 'u' it is not rendered
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Version

v1.58.2

### Expected Behavior

Any label name with value should be added to a sent log without special conditions.

### Actual Behavior

in `ClusterLogDestination`, If you set an `extraLabels` label name to start with `u`, it is missing from a sent log message:

```
spec:
  extraLabels:
    userId: '{{ userId }}'
```

But, if you escape it, it works:
```
spec:
  extraLabels:
    \userId: '{{ userId }}'
```
Produced log file:
```
{
  "userId": "123-123",
  ...
}
```

**Could be more special characters that should be escaped :** https://vector.dev/docs/reference/vrl/expressions/#regular-expression-characteristics

### Steps To Reproduce

1. Default cluster
2. Create `ClusterLogDestination` with `extraLabels` name that starts with `u` and `clusterloggingconfig` for your app with json logs.
3. Catch rendered logs with `vector tap transform/destination/<clusterlogdestination_name>/00_extra_fields`
4. Your added label is missing

### Additional Information

https://github.com/deckhouse/deckhouse/blob/main/modules/460-log-shipper/hooks/internal/vector/transform/extra_fields.go
Probable issue in this function:
[https://github.com/deckhouse/deckhouse/blob/0edd6458a214ce5c5b11e25d3293c9854051ff[…]460-log-shipper/hooks/internal/vector/transform/extra_fields.go](https://github.com/deckhouse/deckhouse/blob/0edd6458a214ce5c5b11e25d3293c9854051ffde/modules/460-log-shipper/hooks/internal/vector/transform/extra_fields.go#L176-L188) 

Applies here, but something is not escaped:
[https://github.com/deckhouse/deckhouse/blob/0edd6458a214ce5c5b11e25d3293c9854051ff[…]460-log-shipper/hooks/internal/vector/transform/extra_fields.go](https://github.com/deckhouse/deckhouse/blob/0edd6458a214ce5c5b11e25d3293c9854051ffde/modules/460-log-shipper/hooks/internal/vector/transform/extra_fields.go#L95)

### Logs

_No response_

### PR DESCRIPTION
**Pull request by ipaqsa**
## Description
Provides extraLabels validation.

## Why do we need it, and what problem does it solve?
Closes #7740
There is a problem related to the fact that you can use reserved keys for extraLabels, and this causes redefinition and unexpected behavior. Adding validation prevents this.

## What is the expected result?
parsed_data cannot be used as a key in extraLabels.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: log-shipper
type: feature 
summary: Add extraLabels validation.
```

### CODE REVIEW

**Comment by meatuses** [comment]
what about other reserved keys? https://deckhouse.io/documentation/v1/modules/460-log-shipper/cr.html#clusterlogdestination-v1alpha1-spec-extralabels

**Comment by ipaqsa** [comment]
> what about other reserved keys? https://deckhouse.io/documentation/v1/modules/460-log-shipper/cr.html#clusterlogdestination-v1alpha1-spec-extralabels

They cannot cause unexpected behavior, and `parsed_data` can. `parsed_data ` consists of other fields, and they use it.
if users want to override other labels, there will be no problems.

File: modules/460-log-shipper/crds/cluster-log-destination.yaml [hunk 1]
```diff
@@ -660,6 +660,9 @@ spec:
                       app_info: '{{ app }}'
                       array_member: '{{ array[0] }}'
                       symbol_escating_value: '{{ pay\.day }}'
+                  x-kubernetes-validations:
+                    - message: parsed_data cannot be used as a key in extraLabels, it's reserved
```
Comment (yalosev) [inline modules/460-log-shipper/crds/cluster-log-destination.yaml:664]: I think we have to exclude all `system` words here
https://deckhouse.io/documentation/v1/modules/460-log-shipper/cr.html#clusterlogdestination-v1alpha1-spec-extralabels

**Comment by ipaqsa** [comment]
Screenshot 2024-04-11 at 6 06 09 PM

**Review by yalosev — APPROVED** [review]

**Pull request merged**